lib/mix/tasks/verify_build.ex

```diff
@@ -0,0 +1,73 @@
+defmodule Mix.Tasks.VerifyBuild do
+  @moduledoc """
+  Run's our build verification suite concurrently, and locally, to test before hitting CI Server.
+  """
+
+  use Mix.Task
+
+  # Thanks for task timeout dialyzer
+  @task_timeout 120_000
+  @ci_tasks [:credo, :dialyzer, :tests]
+
+  @shortdoc "Runs tests, Credo, and Dialyzer."
+
+  def run(_) do
+    @ci_tasks
+    |> Enum.map(fn task -> Task.async(fn -> run_check(task) end) end)
+    |> Enum.map(fn pid -> Task.await(pid, @task_timeout) end)
```

Comment:
This is a great use of Elixir's concurrency and the Erlang scheduler, because Dialyzer can take forever even with the PLT.